newsroom/auth_server/models.py

```diff
@@ -0,0 +1,57 @@
+# # This file is part of Superdesk.
+# #
+# # Copyright 2019 Sourcefabric z.u. and contributors.
+# #
+# # For the full copyright and license information, please see the
+# # AUTHORS and LICENSE files distributed with this source code, or
+# # at https://www.sourcefabric.org/superdesk/license
+
+import logging
+import bcrypt
+from bson import ObjectId
+from bson.errors import InvalidId
+from authlib.oauth2.rfc6749 import ClientMixin
+import superdesk
+
+logger = logging.getLogger(__name__)
+# client_id to OAuth2Client instance map
+
+
+class OAuth2Client(ClientMixin):
+    def __init__(self, data):
+        self._id = data["_id"]
+        self.pwd_hash = data["password"]
+
+    @property
+    def client_id(self):
+        return str(self._id)
+
+    def check_token_endpoint_auth_method(self, method):
+        return method == "client_secret_basic"
```

Comment:
can we make this work also for `client_secret_post` ?
https://docs.authlib.org/en/latest/flask/2/grants.html#client-credentials-grant